implementations/c/include/ockam/handshake.h

```diff
@@ -0,0 +1,124 @@
+/**
+ ********************************************************************************************************
+ * @file    handshake.h
+ * @brief   Interface functions for establishing a secure channel and encrypting/decrypting messages
+ ********************************************************************************************************
+ */
+
+#ifndef OCKAM_HANDSHAKE_H
+#define OCKAM_HANDSHAKE_H
+
+
+/*
+ ********************************************************************************************************
+ *                                             INCLUDE FILES                                            *
+ ********************************************************************************************************
+ */
+#include <stdlib.h>
+#include "ockam/error.h"
+#include "ockam/vault.h"
+#include "ockam/transport.h"
+
+/*
+ ********************************************************************************************************
+ *                                                DEFINES                                               *
+ ********************************************************************************************************
+ */
+
+#define KEY_SIZE 32
+#define SHA256_SIZE 32
+
+/*
+ ********************************************************************************************************
+ *                                               DATA TYPES                                             *
+ ********************************************************************************************************
+ */
+
+/**
+ * XX_HANDSHAKE - the handshake structure is passed to all handshake functions.
+ */
+typedef struct  {
+
+	uint64_t    nonce;
+	uint8_t     s[KEY_SIZE];
```

Comment:
Do these field names refer to variable names in a paper or other document? Can we either link to that resource in a comment, or change the names to be more descriptive?